sdk/core/azure-core/azure/core/serialization.py

```diff
@@ -138,12 +138,18 @@ def _is_readonly(p):
 class AzureJSONEncoder(JSONEncoder):
     """A JSON encoder that's capable of serializing datetime objects and bytes."""
 
+    def __init__(self, *args, exclude_readonly: bool = False, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.exclude_readonly = exclude_readonly
+
     def default(self, o):  # pylint: disable=too-many-return-statements
         if _is_model(o):
-            readonly_props = [
-                p._rest_name for p in o._attr_to_rest_field.values() if _is_readonly(p)
-            ]
-            return {k: v for k, v in o.items() if k not in readonly_props}
+            result = dict(o.items())
```

Comment:
Why are we creating the dict and then pop:ping read-only properties? This seems less efficient than filtering them out before adding them to the new dict....

Reply:
Oh, the logic is for easily deal with `exclude_none`. Since we decide to add it later, I changed back the code.